### PR TITLE
chore(broker-core): migrate msg subscription handling to atomix

### LINKED ISSUE #1936: Migrate SubscriptionCommandSender to Atomix communication service
As one of the first step to remove the zeebe transport module we need to migrate the `SubscriptionCommandSender`, which uses the transport to open and correlation messages subscriptions.

blocks #1925

### PR DESCRIPTION
**Pull request by ChrisKujawa**
closes #1936

### CODE REVIEW

**Comment by Zelldon** [comment]
@menski as discussed I adjusted the PR, please have a look or simply merge :sweat_smile:

File: broker-core/src/main/java/io/zeebe/broker/transport/TransportComponent.java [hunk 1]
```diff
@@ -69,7 +67,6 @@ private void createClientTransports(final SystemContext context) {
     final NetworkCfg networkCfg = brokerCfg.getNetwork();
     final int nodeId = brokerCfg.getCluster().getNodeId();
     final SocketAddress managementEndpoint = networkCfg.getManagement().toSocketAddress();
-    final SocketAddress subscriptionEndpoint = networkCfg.getSubscription().toSocketAddress();
```
Comment (deepthidevaki) [inline broker-core/src/main/java/io/zeebe/broker/transport/TransportComponent.java]: We can also remove subscription from the configuration. It is not needed anymore?

**Review by deepthidevaki — CHANGES_REQUESTED** [review]
Looks good. I have only a suggestion to clean up the configuration together with these changes.

File: broker-core/src/main/java/io/zeebe/broker/transport/TransportComponent.java (hunk 1 above)
Comment (Zelldon) [reply broker-core/src/main/java/io/zeebe/broker/transport/TransportComponent.java]: Yeah agree can do. I thought we will do this in https://github.com/zeebe-io/zeebe/issues/1928

**Pull request merged**